tests/suite/cli_rustup.rs

```diff
@@ -1366,6 +1366,27 @@ fn add_component() {
     });
 }
 
+#[test]
```

Comment:
Do we have a test case testing the "sad path" of the story?

Reply:
More tests added by [d6ef648](https://github.com/rust-lang/rustup/pull/3601/commits/d6ef648f01549dd47dc11d2a93446cccd2b003c6).